ISPN-11504 update docs with segment column name for jdbc cache stores

oraNod opened the pull request:
https://issues.redhat.com/browse/ISPN-11504 

Needs backport to 10.1.x

ryanemerson commented on documentation/src/main/asciidoc/topics/code_examples/PersistenceJdbcStringBasedStore.java:
```diff
@@ -11,6 +11,7 @@
          .idColumnName("ID_COLUMN").idColumnType("VARCHAR(255)")
          .dataColumnName("DATA_COLUMN").dataColumnType("BINARY")
          .timestampColumnName("TIMESTAMP_COLUMN").timestampColumnType("BIGINT")
+         .segmentColumnName("SEGMENT_COLUMN").segmentColumnType("BIGINT")
```
This should probably just be INT as well.

ryanemerson commented on documentation/src/main/asciidoc/topics/con_write_through.adoc:
```diff
@@ -1,10 +1,14 @@
-are updated simultaneously, which ensures that the cache store is always
-consistent with the cache.
+are updated simultaneously, which ensures that the cache store contains all
+entries that are in-memory.
```
The original wording is more accurate. If passivation is enabled, then not all entries will be stored in-memory.

wfink replied:
@ryanemerson @oraNod 
I meant to change or explain the word "cache" as for me "cache" means the complete clustered cache. But in case of a local cache-store (like file) it contains only the entries where this node is the owner.
But maybe this is too technical thinking and might confuse other users

ryanemerson commented on documentation/src/main/asciidoc/topics/config_examples/persistence_example_jdbc.xml:
```diff
@@ -1,14 +1,14 @@
          <id-column name="ID_COLUMN" type="VARCHAR(255)" />
          <data-column name="DATA_COLUMN" type="BINARY" />
          <timestamp-column name="TIMESTAMP_COLUMN" type="BIGINT" />
+         <segment-column name="SEGMENT_COLUMN" type="BIGINT" />
```
INT

ryanemerson commented on documentation/src/main/asciidoc/topics/ref_cache_store_jdbc.adoc:
```diff
@@ -19,5 +19,5 @@ shared.
-`JdbcStringBasedStore` supports segmentation and uses an additional column in
-the database table to represent the segment to which an entry belongs.
+`JdbcStringBasedStore` uses segmentation by default and requires an additional
+column in the database table to represent the segments to which entries belong.
```
Maybe lose "additional" as it's no longer optional by default.

ryanemerson commented on documentation/src/main/asciidoc/topics/ref_cache_store_sfs.adoc:
```diff
@@ -1,14 +1,17 @@
-Single File cache stores, `SingleFileStore`, persist data to a single file and maintains an in-memory index of keys and their values in that file.
+Single File cache stores, `SingleFileStore`, persist data to file. {brandname}
+also maintains an in-memory index of keys and their values in the store. By
+default, Single File cache stores are segmented, which means that {brandname}
```
The second sentence is not that clear imo. The store maintains an in-memory index of keys, whilst key/values are stored in the file.

ryanemerson commented on documentation/src/main/asciidoc/topics/con_write_behind.adoc:
```diff
@@ -26,6 +26,15 @@ write to the cache and the write to the cache store is as small as possible.
+* If `fail-silently="true"` then {brandname} logs WARN messages and rejects
+write operations.
+* If `fail-silently="false"` then {brandname} throws exceptions and rejects
+write operations.
```
This second sentence needs more information. This is what I was clarifying for Wolf. If fail-silently is false, the exceptions are only thrown once the modification queue size becomes full or if  the underlying store is known to be unavailable when a write operation is attempted. For example if the queue size is 100, during the time it takes for the store to be detected as unavailable, it's possible for 99 write operations to be queued without an exception being thrown. If the server is then kill before the store becomes available again, those 99 operations will be lost as they were never persisted.

oraNod commented on documentation/src/main/asciidoc/topics/con_store_migrator.adoc:
```diff
@@ -14,3 +14,9 @@ the target cache.
+====
+`StoreMigrator` migrates non-segmented cache stores to segmented cache stores
+only. It is not currently possible to migrate segmented cache stores.
+====
```
@ryanemerson hey does this sound right? not sure if we need to add some guidance on what to do if you have a segmented cache store with an older version, such as 9.4. I think it's probably not an issue for RHDG given that segmentation was not supported with most cache stores, if any. but could there be a case where users in 9.4 could have a segmented cache store and no path to migrate data to 10+?

ryanemerson replied:
@wburns When were segmented stores added? 9.4.x or 10.x?

wburns replied:
It was added in 9.4.x. The sentence here is incorrect. The problem is you can't migrate from a segmented store to either different number of segment store or non segmented.

oraNod replied:
thanks for clarifying @wburns!

ryanemerson commented:
@oraNod You're going to love me for this, but I think the write-through/behind changes should be broken into a separate commit as they're not related to ISPN-11504.

oraNod commented:
> @oraNod You're going to love me for this, but I think the write-through/behind changes should be broken into separate commit as they're not related to ISPN-11504.



> @oraNod You're going to love me for this, but I think the write-through/behind changes should be broken into separate commit as they're not related to ISPN-11504.

@ryanemerson that's a fair point. tbh I shouldn't have added them to this PR but I was trying to reduce the amount of open doc PRs. I do agree that it's better to keep the commits clean and bound to more specific areas. I've fixed it up now and added write through and write behind to a separate PR. thanks!

ryanemerson commented:
@oraNod I just meant another commit in this PR tbh, but all's good
